causallearn/search/HiddenCausal/GIN/Test_GIN_Condition.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import pandas as pd
+import causallearn.search.HiddenCausal.GIN.FisherTest as FisherTest
+import causallearn.search.HiddenCausal.GIN.Mutual_Information as ID
+
+
+
+def GIN(X, Z, data, test_function, alpha = 0.05):
+    '''
+    Generalized Independent Noise Condition Test
+    Parameters
+    ----------
+    data : numpy ndarray
+           data set
+    X, Z : list
+
+    test_function : str, default='kci'
+        the name of the independence test being used
+    alpha : float, default=0.05
+        desired significance level of independence tests (p_value) in (0,1)
+    Returns
+    -------
+    Boolean : True or False
+    '''
+    omega = getomega(data, X, Z)
+    tdata = data[:,X]
+    result = np.dot(tdata, omega)[:,None]
+    for i in Z:
+        temp = np.array(data[:, [i]])
+        pval = test_function(result, temp)
+        if pval > alpha:
+            flag = True
+        else:
+            flag = False
+
+        if not flag:
+            return False
+
+    return True

```

Comment:
I remembered I commented on this before? 

```suggestion
        if pval <= alpha:
            return False
    return True

```